reactivesocket-client/src/main/java/io/reactivesocket/client/LoadBalancer.java

```diff
@@ -481,17 +470,29 @@ public synchronized String toString() {
     }
 
     @Override
-    public synchronized void close() throws Exception {
-        // TODO: have a `closed` flag?
-        factoryRefresher.close();
-        activeFactories.clear();
-        activeSockets.forEach(rs -> {
-            try {
-                rs.close();
-            } catch (Exception e) {
-                logger.warn("Exception while closing a ReactiveSocket", e);
-            }
-        });
+    public Publisher<Void> close() {
+        return s -> {
+            closeNotifier.onComplete();
```

Comment:
I think it would be simpler and/or easier to read to do the synchronized block first, then notify the subscriber of closeNotifier.